core/src/ast/data.rs

```diff
@@ -122,19 +122,14 @@ impl<V: FromVariant, F: FromField> Data<V, F> {
         match *body {
             syn::Data::Enum(ref data) => {
                 let mut items = Vec::with_capacity(data.variants.len());
```

Comment:
This can be made even shorter, using `filter_map(|v| errors.handle(v)).collect()` to make `items` no longer `mut`.